### PR TITLE
bpo-29613: Added support for SameSite cookies

### PR DESCRIPTION
**Pull request by alex**
This is a rebase of #214.

Closes #214

https://bugs.python.org/issue29613

### CODE REVIEW

File: Doc/library/http.cookies.rst [hunk 1]
```diff
@@ -153,6 +159,9 @@ Morsel Objects
       :attr:`~Morsel.coded_value` are read-only.  Use :meth:`~Morsel.set` for
       setting them.
 
+   .. versionchanged:: 3.8
+      Added support for :attr:`samesite` attribute.
```
Comment (timgraham) [inline Doc/library/http.cookies.rst]: for the

File: Doc/library/http.cookies.rst [hunk 2]
```diff
@@ -137,11 +137,17 @@ Morsel Objects
    * ``secure``
    * ``version``
    * ``httponly``
+   * ``samesite``
 
    The attribute :attr:`httponly` specifies that the cookie is only transferred
    in HTTP requests, and is not accessible through JavaScript. This is intended
    to mitigate some forms of cross-site scripting.
 
+   The attribute :attr:`samesite` specifies that browser is not allowed to send the
```
Comment (timgraham) [inline Doc/library/http.cookies.rst]: "the browser"

File: Doc/library/http.cookies.rst [hunk 3]
```diff
@@ -137,11 +137,17 @@ Morsel Objects
...
+   cookie along with cross-site requests. This help to mitigate CSRF attacks. Valid
```
Comment (timgraham) [inline Doc/library/http.cookies.rst]: help -> helps

File: Doc/library/http.cookies.rst [hunk 4]
```diff
@@ -137,11 +137,17 @@ Morsel Objects
...
+   values for this attribute are "Strict" and "Lax".
+
+
```
Comment (timgraham) [inline Doc/library/http.cookies.rst]: chop extra blank line

File: Lib/test/test_http_cookies.py [hunk 5]
```diff
@@ -121,6 +121,19 @@ def test_set_secure_httponly_attrs(self):
         self.assertEqual(C.output(),
             'Set-Cookie: Customer="WILE_E_COYOTE"; HttpOnly; Secure')
 
+    def test_samesite_attrs(self):
+        samesite_values = ['Strict', 'Lax']
```
Comment (timgraham) [inline Lib/test/test_http_cookies.py]: Might add tests for 'strict' and 'lax' as the values are case-insensitive from what I read.

**Comment by alex** [comment]
Thanks! Feedback addressed

**Review by dstufft — APPROVED** [review]

**Comment by alex** [comment]
@akash0x53 I would encourage you to apply for a Google Patch Reward for your work on this: https://www.google.com/about/appsecurity/patch-rewards/

**Pull request merged**